Add electric parts management when order

bonidier opened the pull request:
Hi @bittner,

Following issue #8, about my part no. 3 (EV3 elements),

For sure Lego don't allow to order electric parts from replacement URL (and don't list all of them !),
So I've scripted my idea, to print their respective Lego shop URL.

FYI, on "replacementparts URL" when you choose set 31313, and select part 6128869, a notice suggest the user to order from Lego shop  :-)

Regards !

bonidier committed:
Add electric parts management when order
- create electric parts CSV file to map Lego's ElementID to standalone set ID)
- print link to order set for electic parts if not found on Lego's replacementparts URL

global :
- join python script's path to default raw-data files

bittner commented on lego-mindstorms-pieces.py:
```diff
@@ -340,9 +374,16 @@ def order(shop=None, browser=None, lego_set=None, order_list=None, username=None
-                set=lego_set))
-            not_in_set_counter += 1
+
+            if is_electric_part(part_no) is True:
```
Remove the `is True`. This will make it more pythonic.

bittner commented on lego-mindstorms-pieces.py:
```diff
@@ -161,10 +164,40 @@ def missing(omitted_set, datafile):
+    """
+
+    wanted_part_no = str(wanted_part_no)
+    datafile = os.path.join(SCRIPT_PATH, 'raw-data', 'Electric Parts.csv')
```
Is it important for you to have a blank on the filename? I'd suggest to use a dash instead, otherwise. Just to avoid typical for system problems.

Same for mixed case versus all small-caps.

bonidier replied:
Absolutly not important. The raw-data/ content have a mixed filename standard ("Lego Mindstorm*.csv" vs "Brickset-inventory*.csv"), So I'd made a random choice. 
Go for first letter to upper case, and lower-with-dash for the rest :-)

bittner commented on lego-mindstorms-pieces.py:
```diff
@@ -161,10 +164,40 @@ def missing(omitted_set, datafile):
+
+            if wanted_part_no == partno:
+                exit_code = True
+                if print_set_link is True:
```
Remove the`is True`. Makes the code more pythonic.

bittner replied:
(BTW, doesn't flake8 complain? Oh no, only Pylint would, probably.)

bonidier replied:
You can be surprised if I admit not being a Python developer :-)

bittner replied:
> You can be surprised if I admit not being a Python developer :-)

I hope you enjoy _becoming_ a Python developer!  :smiley:

bittner commented on lego-mindstorms-pieces.py:
```diff
@@ -161,10 +164,40 @@ def missing(omitted_set, datafile):
+            if wanted_part_no == partno:
+                exit_code = True
+                if print_set_link is True:
+                    print("#{partno} : standalone set URL "
```
No blank before the colon, please! (This is French spelling only)

bonidier replied:
my bad, yet another french bug I should improve

bittner reviewed (changes requested):
A few small changes, would be nice, almost ready to merge.

bonidier committed:
code review - more standardized practice
- rework is_electric_part() conditions
- rename CSV filename for electric parts, avoid blank
- typo fix

bittner commented on lego-mindstorms-pieces.py:
```diff
@@ -161,10 +164,40 @@ def missing(omitted_set, datafile):
+    with open(datafile) as f:
+        data_lines = f.readlines()[1:]
+        for line in data_lines:
+            (partno, legoid, legoshop_set) = line.split('\t')
```
Oh, yet another one: The brackets are strictly not needed. For unpacking you can write:
```python
foo, bar, baz = my_triple_value
```

bonidier replied:
Indeed, it work without brackets.
I'd grabbed it from your code in missing() and parse() functions ;-)

bittner commented on lego-mindstorms-pieces.py:
```diff
@@ -340,9 +374,16 @@ def order(shop=None, browser=None, lego_set=None, order_list=None, username=None
-            not_in_set_counter += 1
+
+            if is_electric_part(part_no):
+                print("For Lego, Electric part are not in set #{set}, see note at the end.".format(
```
I don't fully understand the sentence. Do you intend to say, The LEGO Group treats electric parts separately, as not being part of the set?

Can you re-phrase this a bit? I would guess the sentence is currently French ("Pour LEGO, c'est ..."), n'est-ce pas?

Please also write "LEGO" all-caps (as on the company website) and "electric" all small caps. Merci !

bittner reviewed (approved):
Pretty good for not being a Python developer, well done!

As soon as you address the two comments I added today I'll merge your contribution.

bonidier committed:
re-phrase sentence about electric part

bittner commented:
Merci beaucoup !
